src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -119,6 +119,8 @@ public IHostBuilder ConfigureContainer<TContainerBuilder>(Action<HostBuilderCont
 
         /// <summary>
         /// Run the given actions to initialize the host. This can only be called once.
+        /// Adds host application lifetime service to the host.
+        /// The application will run until interrupted or until <see cref="M:IHostApplicationLifetime.StopApplication()" /> is called.
```

Comment:
```suggestion
```

I don't think this line makes sense on the `Build()` method.

Reply:
I would very much appreciate having this explained somewhere though.  I have put it here because this is what causes this behaviour.  Where would you expect it to appear instead?

Reply:
> I have put it here because this is what causes this behaviour. Where would you expect it to appear instead?

I would expect it on `Start` or `Run`. If you just call `Build()` without calling `Start`, the application won't run until interrupted.

Reply:
But you can also explicitly remove the lifetime service using [RemoveAll](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.extensions.servicecollectiondescriptorextensions.removeall#Microsoft_Extensions_DependencyInjection_Extensions_ServiceCollectionDescriptorExtensions_RemoveAll__1_Microsoft_Extensions_DependencyInjection_IServiceCollection_) and if you do that, it will not keep the application running, so the documentation for [StartAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihost.startasync#Microsoft_Extensions_Hosting_IHost_StartAsync_System_Threading_CancellationToken_) would have to say *The default implementation also includes running the [ApplicationLifetime](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.internal.applicationlifetime) that blah blah blah unless you have explicitly removed it in [Build](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.hostbuilder.build#Microsoft_Extensions_Hosting_HostBuilder_Build)*, which is possible but sounds weird without having it documented on [Build](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.hostbuilder.build#Microsoft_Extensions_Hosting_HostBuilder_Build) itself.

Reply:
OK.  I guess a remark at [StartAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihost.startasync#Microsoft_Extensions_Hosting_IHost_StartAsync_System_Threading_CancellationToken_) to that effect would be appropriate too?  Because it does not make sense to mention the service if it is not required.

Reply:
I'm not sure I understand the question. I don't think we need to document that if you remove all the Lifetime services it will throw, as I don't see that as a common scenario for someone to do.

Reply:
I can move the mention of the lifetime service to [StartAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihost.startasync#Microsoft_Extensions_Hosting_IHost_StartAsync_System_Threading_CancellationToken_) and if you had removed it, you would get an undocumented exception at run time.  Does that sound OK?

Reply:
I don't think we need to document that if you remove all the Lifetime services it will throw, as I don't see that as a common scenario for someone to do.

I think it would be reasonable to add to the StartAsync document that "The application will run until interrupted or until <see cref="M:IHostApplicationLifetime.StopApplication()" /> is called." Today it just says "Starts the program." which isn't that helpful.